sdks/cpp/common/include/Path.h

```diff
@@ -100,9 +100,14 @@ class Path {
      *
      * @return number of segments
      */
-    inline Index size() const { return cend() - front_; }
+    inline Index size() const { return segments_.size() - frontIdx_; }
```

Comment:
I'm curious to know why you switched from using iterators to an index.

Reply:
There was a bug that whenever a Path was copied/moved the front_ iterator still pointed to the old location. Using an index instead of an iterator seemed like the easiest way to solve this problem.

Reply:
I saw you'd made a copy of a Path object somewhere and wanted to talk about that with you today. Looking forward to it!